src/main/java/com/zendesk/maxwell/producer/partitioners/MaxwellKafkaPartitioner.java

```diff
@@ -45,7 +52,14 @@ static protected String getPrimaryKey(RowMap r) {
 		return r.pkAsConcatString();
 	}
 
+	public void setPartitionColumns(String csvColumns) {
+		this.partitionColumns = Arrays.asList(csvColumns.split(","));
+	}
+
 	public int kafkaPartition(RowMap r, int numPartitions) {
-		return Math.abs(hashFunc.hashCode(provider.getHashString(r)) % numPartitions);
+		if ( this.provider instanceof HashStringColumn )
+			return Math.abs(hashFunc.hashCode(((HashStringColumn) provider).getHashString(r, this.partitionColumns)) % numPartitions);
+		else
+			return Math.abs(hashFunc.hashCode(provider.getHashString(r)) % numPartitions);
```

Comment:
this code could be cleaner; I think it would be preferable to put `partitionColumns` in the initializers for both `MaxwellKafkaPartitioner` and `HashStringColumn`